api/files/views.py

```diff
@@ -326,8 +330,12 @@ def get_node(self):
 
     # overrides RetrieveAPIView
     def get_object(self):
-        return self.get_file()
+        user = get_user_auth(self.request).user
+
+        if self.request.GET.get('giveGuid', False) and self.get_node().has_permission(user, 'admin'):
```

Comment:
We don't typically camel-case query params in APIv2. How about just `guid` or `create_guid`?